src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,39 @@
-import React from 'react';
+import { ChangeEvent } from 'react';
+import { StatusFilter } from '../../types/StatusFilter';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as filterActions } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const filterQuery = useAppSelector(state => state.filter);
+  const dispatch = useAppDispatch();
+
+  const changeFilterStatus = (
+    event: ChangeEvent<HTMLSelectElement>,
+  ) => dispatch(
+    filterActions.setTodoFilter(event.target.value as StatusFilter),
+  );
+
+  const changeFilterQuery = (
+    event: ChangeEvent<HTMLInputElement>,
+  ) => dispatch(
+    filterActions.setTodoQuery(event.target.value),
+  );
+
+  const clearQuery = () => dispatch(
+    filterActions.clearTodoQuery(),
+  );
+
   return (
-    <form
-      className="field has-addons"
-      onSubmit={event => event.preventDefault()}
-    >
+    <form className="field has-addons">
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
-            <option value="all">All</option>
-            <option value="active">Active</option>
-            <option value="completed">Completed</option>
+          <select
+            data-cy="statusSelect"
+            onChange={changeFilterStatus}
+          >
+            <option value={StatusFilter.All}>All</option>
+            <option value={StatusFilter.ACTIVE}>Active</option>
+            <option value={StatusFilter.COMPLETED}>Completed</option>
```

Comment:
You can iterate through the filters.

Lets use `Object.entries(([key, value]) => ())`.